src/main/java/net/dries007/tfc/world/placement/OnTopPlacement.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.world.placement;
+
+import java.util.Random;
+import java.util.stream.Stream;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.levelgen.blockpredicates.BlockPredicate;
+import net.minecraft.world.level.levelgen.placement.PlacementContext;
+import net.minecraft.world.level.levelgen.placement.PlacementModifier;
+import net.minecraft.world.level.levelgen.placement.PlacementModifierType;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+public class OnTopPlacement extends PlacementModifier
```

Comment:
This whole class seems designed for an overly specific use case I'm not even sure we need given it's used once in a feature I noted above seems redundant.